Modules/Scripted/DICOMLib/Testing/DICOMBrowserTest.py

```diff
@@ -0,0 +1,120 @@
+
+import os, logging
+import unittest
+import qt
+import slicer
+
+class DICOMBrowserVolumeNaming(unittest.TestCase):
+  def setUp(self):
+    pass
+
+  def delayDisplay(self,message,msec=1000):
+    """This utility method displays a small dialog and waits.
+    This does two things: 1) it lets the event loop catch up
+    to the state of the test so that rendering and widget updates
+    have all taken place before the test continues and 2) it
+    shows the user/developer/tester the state of the test
+    so that we'll know when it breaks.
+    """
+    print(message)
+    self.info = qt.QDialog()
+    self.infoLayout = qt.QVBoxLayout()
+    self.info.setLayout(self.infoLayout)
+    self.label = qt.QLabel(message,self.info)
+    self.infoLayout.addWidget(self.label)
+    qt.QTimer.singleShot(msec, self.info.close)
+    self.info.exec_()
+
+  def runTest(self):
+    self.test_LoadDICOMScalarVolume()
+
+  def test_LoadDICOMScalarVolume(self):
+    """
+    Load a DICOM file with default and customized volume name template, and check that in both cases the name of
+    the volume loaded is ok.
+    The plugin used is DICOMScalarVolumePlugin
+    """
+    # Get the data
+    selfTestDir = slicer.app.temporaryPath + '/DICOMBrowserVolumeNaming'
+    if not os.access(selfTestDir, os.F_OK):
+      os.mkdir(selfTestDir)
+
+    urlDownload = 'http://slicer.kitware.com/midas3/download/item/119940'
+    fileName = 'DicomTestImage00001.dcm'
+    filePath = os.path.join(selfTestDir, fileName)
+    if not os.path.isfile(filePath):
+      import urllib
+      self.delayDisplay("Downloading DICOM data")
+      urllib.urlretrieve(urlDownload, filePath)
+
+    # Load the DICOM file via the DICOM Scalar Volume
+    plugin = slicer.modules.dicomPlugins["DICOMScalarVolumePlugin"]()
+    loadables = plugin.examineForImport([[filePath]])
+    plugin.load(loadables[0])
+    # Check the volume is using default template (seriesNumber: seriesDescription)
+    self.assertIsNotNone(slicer.util.getNode("6: Unknown"))
+
+    # Open a new copy of the volume using a different template (Patient name, series number)
+    plugin.loadableCache.clear()  # Clear cache
+    plugin.volumeNameTagsTemplate = "@PatientName@: @0020,0011@"
+    loadables = plugin.examineForImport([[filePath]])
+    plugin.load(loadables[0])
+    slicer.util.getNode("RANDO^PROSTATE: 6")
+
+    self.delayDisplay("Test passed!")
+
+  def tearDown(self):
+    slicer.mrlmlScene.Clear(0)
+
+#
+# DICOM Browser tests
+#
+class DICOMBrowserTest:
+  """
+  This class is the 'hook' for slicer to detect and recognize the test
+  as a loadable scripted module (with a hidden interface)
+  """
+  def __init__(self, parent):
+    parent.title = "DICOMBrowserTest"
+    parent.categories = ["Testing.TestCases"]
+    parent.contributors = ["Jorge Onieva (BWH)"]
+    parent.helpText = """Self tests for the DICOM Browser. So far, no GUI is tested"""
```

Comment:
Adding:
```
parent.dependencies = ['DICOM']
```
will ensure `slicer.dicomDatabase` is set.

See here for more details: https://github.com/Slicer/Slicer/blob/acc9915ce733f2ad8394d9566a73580ad3327cb7/Modules/Scripted/DICOM/DICOM.py#L42-L65